pkg/detectors/twist/twist.go

```diff
@@ -0,0 +1,77 @@
+package twist
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct{}
+
+// Ensure the Scanner satisfies the interface at compile time
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	client = common.SaneHttpClient()
+
+	//Make sure that your group is surrounded in boundry characters such as below to reduce false positives
+	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"twist"}) + `\b([0-9a-f:]{40,47})\b`)
```

Comment:
i think you should include `oauth2:` in the regex pattern. we expect that the token already contains `oauth2:`

Reply:
No we can't include the oauth2: in the regex pattern because token generated using Basic authentication doesn't have oauth2 in it. Please check it here https://developer.twist.com/v3/#login.